sport-moment-nft/contracts/EnglishPremierLeague.cdc

```diff
@@ -0,0 +1,1134 @@
+    //------------------------------------------------------------
+
+    /// Entity Counts
+    ///
+    pub var totalSupply:        UInt64
+    pub var nextSeriesID:       UInt64
+    pub var nextSetID:          UInt64
+    pub var nextTagID:          UInt64
+    pub var nextPlayID:         UInt64
+    pub var nextEditionID:      UInt64
+
+    //------------------------------------------------------------
+    // Internal contract state
+    //------------------------------------------------------------
+
+    /// Metadata Dictionaries
+    ///
+    access(self) let seriesIDByName:    {String: UInt64}
+    access(self) let seriesByID:        @{UInt64: Series}
+    access(self) let setIDByName:       {String: UInt64}
+    access(self) let setByID:           @{UInt64: Set}
+    access(self) let tagByID:           @{UInt64: Tag}
+    access(self) let playByID:          @{UInt64: Play}
+    access(self) let editionByID:       @{UInt64: Edition}
+
+    /// Contract vars
+    pub var royaltyAddress: Address
+
+    //------------------------------------------------------------
+    // Series
+    //------------------------------------------------------------
+
+    /// A public struct to access Series data
+    ///
+    pub struct SeriesData {
+        pub let id: UInt64
+        pub let name: String
+        pub let active: Bool
+
+        /// initializer
+        //
+        init (id: UInt64) {
+            let series = (&EnglishPremierLeague.seriesByID[id] as! &EnglishPremierLeague.Series?)!
+            self.id = series.id
+            self.name = series.name
+            self.active = series.active
+        }
+    }
+
+    /// A top-level Series with a unique ID and name
+    ///
+    pub resource Series {
+        pub let id: UInt64
+        pub let name: String
+        pub var active: Bool
+
+        /// Close this series
+        ///
+        pub fun close() {
+            pre {
+                self.active == true: "series is not active"
+            }
+
+            self.active = false
+
+            emit SeriesClosed(id: self.id)
+        }
+
+        /// initializer
+        ///
+        init (name: String) {
+            pre {
+                !EnglishPremierLeague.seriesIDByName.containsKey(name): "A Series with that name already exists"
+            }
+            self.id = EnglishPremierLeague.nextSeriesID
+            self.name = name
+            self.active = true
+
+            // Cache the new series's name => ID
+            EnglishPremierLeague.seriesIDByName[name] = self.id
+            // Increment for the nextSeriesID
+            EnglishPremierLeague.nextSeriesID = self.id + 1 as UInt64
+
+            emit SeriesCreated(id: self.id, name: self.name)
+        }
+    }
+
+    /// Get the publicly available data for a Series by id
+    ///
+    pub fun getSeriesData(id: UInt64): EnglishPremierLeague.SeriesData {
+        pre {
+            EnglishPremierLeague.seriesByID[id] != nil: "Cannot borrow series, no such id"
```

Comment:
Forgot this earlier, but whenever you have a getter function, you should always return a nil if something goes wrong instead of panicing